### PR TITLE
Setup a provider now has optional required keys

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 834c6a09b89ac96f9d004f07c385718f9eb6f098:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: utils/providers.py [hunk 1]
```diff
@@ -168,6 +171,15 @@ def setup_a_provider(prov_class=None, prov_type=None, validate=True, check_exist
         from cfme.infrastructure.provider import get_from_config
         providers = list_infra_providers()
 
+    final_providers = []
+    for provider in providers:
```
Comment (seandst) [inline utils/providers.py]: Neither of these is the fault of this PR, but:
- the docs say set up a random provider. It'd be good to see some randomness in here,  probably by shuffling the provider list.
- problematic_providers doesn't need to be global'd

Comment (seandst) [inline utils/providers.py]: Ignore the shuffle comment. :)

**Comment by dajoRH** [comment]
Lint report for commit 9d46021f66cf38a5a26095979cbe1c5f1c0631a5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8bad241f56fbc6f09db0dec07cc4329bbf03c2a9:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8c2e20f0f3a78c683af3cb6b180603f82ce70862:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: utils/providers.py [hunk 2]
```diff
@@ -168,6 +171,12 @@ def setup_a_provider(prov_class=None, prov_type=None, validate=True, check_exist
...
+        if all(key in providers_data[provider] for key in required_keys):
```
Comment (mfalesni) [inline utils/providers.py]: (inc generators)

**Pull request merged**